### PR TITLE
xthor: misc enhancements

### PR DESCRIPTION
**Pull request by JigSawFr**
- **New:** Added categories dropping options (enabled, only for TMDB queries, disabled) - For TMDB queries by default
- **Improved:** Removed search parameter when no search term submitted.

### CODE REVIEW

**Comment by JigSawFr** [comment]
Categories are known problems for API if there are too many included in the request. (It will send bad results and duplicated on multiple pages). For TMDB queries, it's more accurate as only matched TMDB id are returned, so results are better (performance too).

File: src/Jackett.Common/Indexers/Xthor.cs [hunk 1]
```diff
@@ -235,6 +236,15 @@ public class Xthor : BaseCachingWebIndexer
             var ConfigMaxPagesBypassForTMDB = new BoolConfigurationItem("Do you want to bypass max pages for TMDB searches ? (Radarr) - Hard limit of 4") { Value = true };
             ConfigData.AddDynamic("maxPagesBypassForTMDB", ConfigMaxPagesBypassForTMDB);
 
+            var ConfigDropCategories = new SingleSelectConfigurationItem("Drop categories requested ?", new Dictionary<string, string>
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Indexers/Xthor.cs:239]: `Drop categories requested ?` > `Drop requested categories`

**Comment by ilike2burnthing** [comment]
I don't have an account to test and am not a C# dev so can only comment generally, but if it's working for you, then this looks fine to me.

Just those two comments and those from the previous PR, pretty much all small things, e.g. we don't have questions in settings for any other indexer, so rephrasing things to avoid them here.

**Comment by JigSawFr** [comment]
@ilike2burnthing thanks for your comments. For other's changes I will make another PR ;)

**Review by ilike2burnthing — APPROVED** [review]

**Pull request merged**